dask/dataframe/io/parquet/core.py

```diff
@@ -694,6 +694,9 @@ def to_parquet(
             "columns=%s" % (str(partition_on), str(list(df.columns)))
         )
 
+    if df.columns.inferred_type not in {"string", "empty"}:
+        raise ValueError("parquet doesn't support non-string column names")
```

Comment:
TIL about `Index.inferred_type` -- thanks @jcrist

Just linking to the corresponding logic over in `pandas` for other readers https://github.com/pandas-dev/pandas/blob/41e423fd09df5817dcfa11f3aea111a5b1a2f98a/pandas/io/parquet.py#L129-L130